From f38c30bd69bf56ea884359923ec10f752fb6e663 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 5 Dec 2012 18:54:17 +0800
Subject: [PATCH] rk29_key: call keys_isr when resume

This is patch for commit cdd3cccbd6665f5d8ea248d510cbafc6cf8380f5
keyboard: If the power button is pressed in suspend state, immediately report event
---
 drivers/input/keyboard/rk29_keys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/keyboard/rk29_keys.c b/drivers/input/keyboard/rk29_keys.c
index 38663abb1efd..1eca1d6a8135 100755
--- a/drivers/input/keyboard/rk29_keys.c
+++ b/drivers/input/keyboard/rk29_keys.c
@@ -514,6 +514,10 @@ static int keys_resume(struct device *dev)
 				disable_irq_wake(irq);
 			}
 		}
+		preempt_disable();
+		if (local_softirq_pending())
+			do_softirq(); // for call resend_irqs, which may call keys_isr
+		preempt_enable_no_resched();
 	}
 
 	ddata->in_suspend = false;
-- 
2.35.3

